### ISSUE TITLE
Google OAuth2 plugin

### ISSUE THREAD

**Issue by azurit** [coreruleset/coreruleset#2232]
### Motivation

Lots of users are reporting problems with Google OAuth2 callback requests as it's `scope` argument usually contains string `.profile`, which is triggering rule `930120`. We already proposed a solution for this (see PRs #1958 and #2222) but i don't think that our core ruleset should contain bypasses for specific software and services.

### Proposed solution

I suggest reworking this into an official plugin, which will contain 3 rules:
 - current rule `930050`
 - new rule `930051` from #2222
 - and, finally, rule similar to this:
```
SecRule TX:GOOGLE_OAUTH2_CALLBACK_DETECTED "@eq 1" \
    "id:930052,\
    phase:2,\
    pass,\
    t:none,\
    nolog,\
    ctl:ruleRemoveTargetById=930120;ARGS:scope"

```

Prototype of this plugin is ready.

**Comment by dune73** [comment]
Linking #2212 where another `.profile` case is documented.

I'm adding this to the agenda for tonight's issue chat.

**Comment by dune73** [comment]
It was decided, that this will be covered in a plugin. Now waiting for said plugin.

**Comment by azurit** [comment]
@dune73 Can you, please, create a new repository for this plugin? Probably with name `google-oauth2-plugin`. Thank you.

**Comment by dune73** [comment]
Of course.

**Comment by dune73** [comment]
https://github.com/coreruleset/google-oauth2-plugin

**Comment by dune73** [comment]
Let me know if you want to change the description (top right) and GH does not grant you permissions.

**Comment by azurit** [comment]
Plugin is ready: https://github.com/coreruleset/google-oauth2-plugin. Closing!
